addons/code-analysis/userscript.js

```diff
@@ -0,0 +1,219 @@
+const functionOpcode = "procedures_definition";
+
+const appearanceCodeTypeList = ["motion", "looks", "sound"];
+
+const getterMap = {
+  motion: (data) => (isNaN(data["motion"]) ? 0 : data["motion"]),
+  looks: (data) => (isNaN(data["looks"]) ? 0 : data["looks"]),
+  sound: (data) => (isNaN(data["sound"]) ? 0 : data["sound"]),
+  events: (data) => (isNaN(data["event"]) ? 0 : data["event"]),
+  control: (data) => (isNaN(data["control"]) ? 0 : data["control"]),
+  sensing: (data) => (isNaN(data["sensing"]) ? 0 : data["sensing"]),
+  operators: (data) => (isNaN(data["operator"]) ? 0 : data["operator"]),
+  variables: (data) => (isNaN(data["data"]) ? 0 : data["data"]),
+  "my-blocks": (data) => {
+    const argument = isNaN(data["argument"]) ? 0 : data["argument"];
+    const procedures = isNaN(data["procedures"]) ? 0 : data["procedures"];
+    return argument + procedures;
+  },
+};
+
+const codeGeneticList = [
+  "motion",
+  "looks",
+  "sound",
+  "events",
+  "control",
+  "sensing",
+  "operators",
+  "variables",
+  "my-blocks",
+];
+
+const codeGeneticTableList = [
+  ["motion", "events", "operators"],
+  ["looks", "control", "variables"],
+  ["sound", "sensing", "my-blocks"],
+];
+
+export default async function ({ addon, msg, console }) {
+  const vm = addon.tab.traps.vm;
+  await new Promise((resolve) => {
+    if (vm.editingTarget) return resolve();
+    vm.runtime.once("PROJECT_LOADED", resolve);
+  });
+  const targets = addon.tab.traps.vm.runtime.targets;
+  const result = analysisCode(targets);
+  const content = await addon.tab.waitForElement("#view > div > div.inner");
+  const projectStatsContainer = document.createElement("div");
+  content.appendChild(projectStatsContainer);
+  projectStatsContainer.className = "flex-row preview-row";
+  const projectStatsContent = document.createElement("div");
+  projectStatsContainer.appendChild(projectStatsContent);
+  projectStatsContent.className = "flex-row project-stats-content";
+  const container = createElement("ps-container", "div");
+  projectStatsContent.appendChild(container);
+  renderStats(addon, container, result);
+}
+
+function analysisCode(targets) {
```

Comment:
I think the function name makes more sense if its `analyseCode`
```suggestion
function analyseCode(targets) {
```

Reply:
ok，I've renamed it.